src/main/java/io/supertokens/thirdparty/ThirdParty.java

```diff
@@ -159,6 +161,10 @@ public static UserPaginationContainer getUsers(Main main,
         return new UserPaginationContainer(resultUsers, nextPaginationToken);
     }
 
+    public static UserInfo[] getUsersByEmail(Main main, @Nonnull GetUsersByEmailQuery query) throws StorageQueryException {
```

Comment:
This function should just take `Main main, @Nonnull String email` as the input. No need to create an abstraction for `GetUsersByEmailQuery`... keep it simple :)